secp256k1_test.go

```diff
@@ -501,6 +501,126 @@ func TestDeterministicSchnorrSignatureGen(t *testing.T) {
 	}
 }
 
+func TestSchnorrPublicKey_IsEqual(t *testing.T) {
+	r := rand.New(rand.NewSource(1))
+	goodPrivateKey := fastGeneratePrivateKey(t, r)
+	goodPublicKey, err := goodPrivateKey.SchnorrPublicKey()
+	if err != nil {
+		t.Fatalf("Failed generating pubkey from: '%s'. '%s'", goodPrivateKey, err)
+	}
+	badPublicKey := SchnorrPublicKey{}
+	if badPublicKey.IsEqual(goodPublicKey) {
+		t.Errorf("Emtpy publickey shouldn't be equal to good one")
+	}
+	if !badPublicKey.IsEqual(&SchnorrPublicKey{}) {
+		t.Errorf("Emtpy publickey should be equal to another empty pubkey")
+	}
+	var nilPubKey *SchnorrPublicKey = nil
+	if nilPubKey.IsEqual(goodPublicKey) {
+		t.Errorf("nil publickey shouldn't be equal to good one")
+	}
+
+	if !nilPubKey.IsEqual(nil) {
+		t.Errorf("two nil pubkeys should be equal")
+	}
+
+	copyGoodPubkey := *goodPublicKey
+	if !copyGoodPubkey.IsEqual(goodPublicKey) {
+		t.Errorf("A pubkey and it's copy should be the same")
+	}
+	goodPrivateKey2 := fastGeneratePrivateKey(t, r)
+	goodPublicKey2, err := goodPrivateKey2.SchnorrPublicKey()
+	if err != nil {
+		t.Fatalf("Failed generating pubkey from: '%s'. '%s'", goodPrivateKey2, err)
+	}
+
+	if goodPublicKey.IsEqual(goodPublicKey2) {
+		t.Errorf("'%s' shouldn't be equal to %s", goodPublicKey, goodPublicKey2)
+	}
+}
+
+func TestSchnorrSignature_IsEqual(t *testing.T) {
+	r := rand.New(rand.NewSource(1))
+	serializedSig := SerializedSchnorrSignature{}
+	n, err := r.Read(serializedSig[:])
+	if err != nil || n != len(serializedSig) {
+		t.Errorf("Failed generating a random signature '%d' '%s' ", n, err)
+	}
+	goodSignature := DeserializeSchnorrSignature(&serializedSig)
+
+	emptySignature := SchnorrSignature{}
+	if emptySignature.IsEqual(goodSignature) {
+		t.Errorf("Emtpy signature shouldn't be equal to good one")
+	}
+	if !emptySignature.IsEqual(&SchnorrSignature{}) {
+		t.Errorf("Emtpy signature should be equal to another empty signature")
+	}
+	var nilSignature *SchnorrSignature = nil
+	if nilSignature.IsEqual(goodSignature) {
+		t.Errorf("nil signature shouldn't be equal to good one")
+	}
+
+	if !nilSignature.IsEqual(nil) {
+		t.Errorf("two nil signatures should be equal")
+	}
+
+	copyGoodSignature := *goodSignature
+	if !copyGoodSignature.IsEqual(goodSignature) {
+		t.Errorf("A signature and it's copy should be the same")
+	}
+
+	serializedSig2 := SerializedSchnorrSignature{}
+	n, err = r.Read(serializedSig2[:])
+	if err != nil || n != len(serializedSig2) {
+		t.Errorf("Failed generating a random signature '%d' '%s' ", n, err)
```

Comment:
Please annotate what n is in the error and remove the whitespace at the end of the line.

Reply:
Same as above